### PR TITLE
[4.0 -> 5.0] Fix non-working cleos set code and set abi commands, and add tests

### PR DESCRIPTION
**Pull request by linh2931**
Cleos `set code` and `set abi` have not worked since `release/3.2`. The root cause is the common code `set_code_callback` for `set contract` and `set code` assumes `contract-dir` argument which is only applicable to `set contract`; and the common code `set_code_callback` for `set abi` and `set code` assumes `contract-dir` argument too.

Merges fix from `release/4.0` into `release/5.0` including [#1897](https://github.com/AntelopeIO/leap/pull/1897).

In addition to the fix, tests are added in this PR for `release/5.0` and `main`.

Resolves https://github.com/AntelopeIO/leap/issues/1868.

### CODE REVIEW

File: tests/cleos_set_code_test.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import random
+from TestHarness import Account, Cluster, TestHelper, Utils, WalletMgr
+
+###############################################################
+# cleos_set_code_test
+#
+# Test Cleos "set code" and "set abi" commands
+#
+###############################################################
```
Comment (heifner) [inline tests/cleos_set_code_test.py:11]: This test is fine, but I think you could have just added it to `nodeos_run_test.py` to save on test run times.

File: tests/cleos_set_code_test.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+Print=Utils.Print
+errorExit=Utils.errorExit
+
+args=TestHelper.parse_args({"--seed"
```
Comment (heifner) [inline tests/cleos_set_code_test.py:16]: Why the `seed` argument?

File: tests/cleos_set_code_test.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                            ,"--dump-error-details","-v","--leave-running"
+                            ,"--keep-logs","--unshared"})
+
+pnodes=1
+topo="mesh"
```
Comment (heifner) [inline tests/cleos_set_code_test.py:21]: The default is `mesh`

**Review by heifner — APPROVED** [review]

File: tests/cleos_set_code_test.py (hunk 1 above)
Comment (linh2931) [reply tests/cleos_set_code_test.py:11]: Thanks! I will change to use `nodeos_run_test.py` to save test setup time.

**Review by spoonincode — APPROVED** [review]

**Review by heifner — APPROVED** [review]

**Pull request merged**